openpype/hosts/houdini/plugins/publish/collect_rop_frame_range.py

```diff
@@ -2,40 +2,103 @@
 """Collector plugin for frames data on ROP instances."""
 import hou  # noqa
 import pyblish.api
+from openpype.lib import BoolDef
 from openpype.hosts.houdini.api import lib
+from openpype.pipeline import OpenPypePyblishPluginMixin
 
 
-class CollectRopFrameRange(pyblish.api.InstancePlugin):
+class CollectRopFrameRange(pyblish.api.InstancePlugin,
+                           OpenPypePyblishPluginMixin):
+
     """Collect all frames which would be saved from the ROP nodes"""
 
+    hosts = ["houdini"]
     order = pyblish.api.CollectorOrder
     label = "Collect RopNode Frame Range"
+    use_asset_handles = True
 
     def process(self, instance):
 
         node_path = instance.data.get("instance_node")
         if node_path is None:
             # Instance without instance node like a workfile instance
+            self.log.debug(
+                "No instance node found for instance: {}".format(instance)
+            )
             return
 
         ropnode = hou.node(node_path)
-        frame_data = lib.get_frame_data(ropnode)
 
-        if "frameStart" in frame_data and "frameEnd" in frame_data:
+        attr_values = self.get_attr_values_from_data(instance.data)
 
-            # Log artist friendly message about the collected frame range
-            message = (
-                "Frame range {0[frameStart]} - {0[frameEnd]}"
-            ).format(frame_data)
-            if frame_data.get("step", 1.0) != 1.0:
-                message += " with step {0[step]}".format(frame_data)
-            self.log.info(message)
+        if attr_values.get("use_handles", self.use_asset_handles):
+            asset_data = instance.context.data["assetEntity"]["data"]
```

Comment:
This should be `instance.data["assetEntity"]` no? Since technically an instance could publish INTO another asset than the current context? what do we want to do with the handles then?

Reply:
Good Point! 

Should I do something similar to this ? 
```python
if  instance.data["asset"] != os.environ["AVALON_ASSET"]
    asset_doc = get_asset_by_name(os.environ["AVALON_PROJECT"], instance.data["asset"])
    asset_data = asset_doc["data"]
    handle_start = asset_data["handleStart"]
    handle_end = asset_data["handleEnd"]
```

Reply:
No - I believe instance data gets `assetEntity` correctly as well. So try that first.

Reply:
It gets `assetEntity` correctly but for the current asset even if the instance is pointing to another asset.

Reply:
That's odd it should be set correctly to `instance.data["assetEntity"]` via the code [here](https://github.com/ynput/OpenPype/blob/745aacea0c7db07b0da408af990a480dff45bc31/openpype/plugins/publish/collect_anatomy_instance_data.py#L59-L116)

Reply:
I'm using  `instance.context.data["assetEntity"]["data"]`
I face error when using `instance.data["assetEntity"]`
```
  File "E:\Ynput\OpenPype\openpype\hosts\houdini\plugins\publish\collect_rop_frame_range.py", line 35, in process
    asset_data = instance.data["assetEntity"]["data"]
KeyError: 'assetEntity'
```

Reply:
> That's odd it should be set correctly to instance.data["assetEntity"] via the code [here](https://github.com/ynput/OpenPype/blob/745aacea0c7db07b0da408af990a480dff45bc31/openpype/plugins/publish/collect_anatomy_instance_data.py#L59-L116)

the code you are pointing to has
```python
order = pyblish.api.CollectorOrder + 0.49
```

while mine has
```python
order = pyblish.api.CollectorOrder
```

Reply:
> I think we should have some sort of dependency graph with plugins and their order which will act as a reference to pick an adequate order value!

The details list in the publisher should be ordered by their `id` at least ;) haha.
Anyway, yes - then I suppose we need to tweak the order here. @iLLiCiTiT thoughts?

Reply:
I though about using more standardized constants as follows 
but I think this is not enough because plugins can be intertwined ! 

```python
PreCollectorOrder = pyblish.api.CollectorOrder - 0.1
CollectorOrder = pyblish.api.CollectorOrder
PostCollectorOrder = pyblish.api.CollectorOrder + 0.1

PreValidatorOrder = pyblish.api.ValidatorOrder - 0.1
ValidatorOrder = pyblish.api.ValidatorOrder
PostValidatorOrder = pyblish.api.ValidatorOrder + 0.1
```

The idea it self is not new, but this is not working for us because (I assume) we didn't keep track of the order of the plugins when we wrote them.

https://github.com/ynput/OpenPype/blob/9aa48e8957de5a4a82fd06d5c8ed4b6feed038d0/openpype/pipeline/publish/constants.py#L4-L7

Reply:
I think there's no particular need to get more granular and it's clear enough to just have `CollectorOrder + x` or `CollectorOrder - x`. The `ValidatePipelineOrder` etc also got less traction because it just didn't matter that much. If something had to run after/before another you'd still offset them anyway.

So I'd say if we can ignore the constants let's do so - keeps things thin.

Reply:
Completelly out of scope of this PR. We need to define orders, but now is not the best time to do so...

Reply:
> I don't know the consequences of that higher order. e.g., CollectHoudiniReviewData with order collector +0.1 reads instance.data["frameStart"] and instance.data["frameEnd"]

As far as I know this plugin should not change `"frameStart"` but `"frameStartHandle"`. In other cases please look if it's possible to move houdini plugins to work properly.

The issue with `CollectAnatomyInstanceData` is that at this moment it must be at that order...

Reply:
Sounds fine with me - just make sure not to use `+ 0.5` but `+ 0.4999` just to ensure we haven't offsetted into the range of Validators?